### PR TITLE
Match More Functions

### PR DESCRIPTION
**Pull request by PokeCodec**

### CODE REVIEW

**Comment by PikalaxALT** [comment]
PR title does not describe work performed

**Comment by PokeCodec** [comment]
@PikalaxALT fixed

File: src/pokedex.c [hunk 1]
```diff
@@ -5084,74 +5084,24 @@ static void sub_80927F0(u8 taskId)
...
 
     for (i = 0; i < d; i++)
     {
-        ((u16 *)VRAM)[15 * 0x400 + c * 32 + i + b] &= 0xFFF;
-        ((u16 *)VRAM)[15 * 0x400 + c * 32 + i + b] |= a << 12;
+        j = *(u16 *)(VRAM + 15 * 0x800 + c * 64 + (b + i)*2);
+        j &= 0xFFF;
+        j |= a << 12;
+
+        *(u16 *)(VRAM + 15 * 0x800 + c * 64 + (b + i)*2) = j;
```
Comment (PikalaxALT) [inline src/pokedex.c:5097]: This looks macro-like

Comment (PokeCodec) [reply src/pokedex.c:5097]: Should I make a macro?

Comment (PokeCodec) [reply src/pokedex.c:5097]: I don't think a macro is needed because macros are for reuse. this is the only time I do this.

File: src/pokedex.c [hunk 2]
```diff
@@ -25,7 +25,7 @@
 #include "trig.h"
 #include "scanline_effect.h"
 #include "ewram.h"
-
+#define VRAM_ADDR(a, b, c) *(u16 *)(BG_VRAM + (15 * 0x800) + (c) * 64 + ((b) + (a))*2)
```
Comment (GriffinRichards) [inline src/pokedex.c:28]: Make this define once, don't duplicate it in every file it's needed

**Comment by PokeCodec** [comment]
Each and every issue addressed!

File: constants/m4a_constants.inc [hunk 3]
```diff
@@ -1,199 +1,183 @@
-	.equiv ID_NUMBER, 0x68736d53
+.equiv ID_NUMBER, 0x68736d53
```
Comment (PikalaxALT) [inline constants/m4a_constants.inc:1]: Revert changes in m4a_constants.inc

File: include/gba/m4a_internal.h [hunk 4]
```diff
@@ -140,55 +140,13 @@ struct SoundChannel
     struct WaveData *wav;
     u32 cp;
     struct MusicPlayerTrack *track;
-    u32 pp;
-    u32 np;
-    u32 d4;
+    void* pp;
```
Comment (PikalaxALT) [inline include/gba/m4a_internal.h:143]: Pointer style inconsistent with line 103

File: include/gba/m4a_internal.h [hunk 5]
```diff
@@ -334,7 +292,47 @@ struct Song
 extern const struct MusicPlayer gMPlayTable[];
 extern const struct Song gSongTable[];
 
+#define MAX_DIRECTSOUND_CHANNELS 12
```
Comment (PikalaxALT) [inline include/gba/m4a_internal.h:295]: Why move this definition?

File: src/battle/anim/psychic.c [hunk 6]
```diff
@@ -516,7 +516,7 @@ static void sub_80DB92C(struct Sprite *sprite)
 {
     REG_BLDALPHA = ((16 - sprite->data[3]) << 8) | sprite->data[3];
 
-    if (--sprite->data[3] == -1)
+    if (sprite->data[3]-- == 0)
```
Comment (PikalaxALT) [inline src/battle/anim/psychic.c:519]: Is this change required to match?

File: src/libs/libagbsyscall.s [hunk 7]
```diff
@@ -5,61 +5,61 @@
 
 	thumb_func_start ArcTan2
 ArcTan2: @ 81E07E0
-	swi 0xA
+	svc 0xA
```
Comment (PikalaxALT) [inline src/libs/libagbsyscall.s:8]: Revert

File: src/palette.c [hunk 8]
```diff
@@ -149,47 +149,43 @@ void ReadPlttIntoBuffers(void)
 bool8 BeginNormalPaletteFade(u32 selectedPalettes, s8 delay, u8 startY, u8 targetY, u16 blendColor)
 {
     u8 temp;
-    register u32 _blendColor asm("r8") = blendColor;
 
-    if (gPaletteFade.active)
+    if (gPaletteFade.active == TRUE)
```
Comment (PikalaxALT) [inline src/palette.c:153]: ?????

Comment (PokeCodec) [reply src/palette.c:153]: if (a == true) is NOT the same as if (a). they generate different code.

File: include/gba/m4a_internal.h (hunk 5 above)
Comment (PokeCodec) [reply include/gba/m4a_internal.h:295]: So it can be grouped together with the PCM_DMA_BUF_SIZE, as both macros are used in conjunction a lot

File: src/battle/anim/psychic.c (hunk 6 above)
Comment (PokeCodec) [reply src/battle/anim/psychic.c:519]: It makes more sense this way

**Comment by PokeCodec** [comment]
Fixed!

**Comment by PokeCodec** [comment]
if (a == true) is NOT the same as if (a). they generate different code. @PikalaxALT
